src/components/pages/PageDecksCreate/PageDecksCreate.jsx

```diff
@@ -0,0 +1,252 @@
+          .where('type', '==', 'Character')
+          .get();
+
+        const newCharacterOptions = cardRefs
+          .docs
+          .map((card) => {
+            const cardData = {
+              id: card.id,
+              ...card.data(),
+            };
+            return { label: cardData.cardTitle, value: cardData.id };
+          });
+
+        setCharacterOptions(newCharacterOptions);
+        setIsLoading(true);
+      } catch (err) {
+        console.error(err);
+      }
+    },
+    [user.uid],
+  );
+
+  const populateCircumstances = useCallback(
+    async () => {
+      try {
+        const cardRefs = await firebase.db
+          .collection('cards')
+          .where('createdBy', '==', user.uid)
+          .where('type', '==', 'Circumstance')
+          .get();
+
+        const newCircumstanceOptions = cardRefs
+          .docs
+          .map((card) => {
+            const cardData = {
+              id: card.id,
+              ...card.data(),
+            };
+            return { label: cardData.cardTitle, value: cardData.id };
+          });
+
+        setCircumstanceOptions(newCircumstanceOptions);
+        setIsLoading(true);
+      } catch (err) {
+        console.error(err);
+      }
+    },
+    [user.uid],
+  );
+
+  const populateConflicts = useCallback(
+    async () => {
+      try {
+        const cardRefs = await firebase.db
+          .collection('cards')
+          .where('createdBy', '==', user.uid)
+          .where('type', '==', 'Conflict')
+          .get();
+
+        const newConflictOptions = cardRefs
+          .docs
+          .map((card) => {
+            const cardData = {
+              id: card.id,
+              ...card.data(),
+            };
+            return { label: cardData.cardTitle, value: cardData.id };
+          });
+
+        setConflictOptions(newConflictOptions);
+        setIsLoading(true);
+      } catch (err) {
+        console.error(err);
+      }
+    },
+    [user.uid],
+  );
+
+  function changeDeck(key, value) {
+    const newDeck = { ...deck };
+    newDeck[key] = value;
+    setDeck(newDeck);
+  }
+
+  async function handleCreateCard() {
+    const newDeck = {
+      name: deck.name,
+      description: deck.description,
+      characterCards: selectedCharacterIds.map((cardId) => (
+        {
+          cardRef: firebase.db.collection('cards').doc(cardId.value),
+          quantity: 1,
+        }
+      )),
+      circumstanceCards: selectedCircumstanceIds.map((cardId) => (
+        {
+          cardRef: firebase.db.collection('cards').doc(cardId.value),
+          quantity: 1,
+        }
+      )),
+      conflictCards: selectedConflictIds.map((cardId) => (
+        {
+          cardRef: firebase.db.collection('cards').doc(cardId.value),
+          quantity: 1,
+        }
+      )),
+      visibility: 'public',
+      createdBy: user.uid,
+    };
+
+    try {
+      await firebase.db.collection('decks').add(newDeck);
+    } catch (err) {
```

Comment:
If this fails you need to give a proper error message. Don't let the user be confused.